src/zenml/models/pipeline_models.py

```diff
@@ -100,14 +101,16 @@ class StepRunModel(DomainModel):
 
     pipeline_run_id: UUID
     parent_step_ids: List[UUID]
+    input_artifacts: Dict[str, UUID]  # mapping from input name to artifact ID
```

Comment:
Which artifact is an output of which step is a 1-to-N relationship defined by the `ArtifactModel.parent_step_id` attribute, so, for internal purposes, we don't necessarily need to store this link in the `StepRunModel` again.

Input artifacts are more tricky since that is an N-to-M relationship, which we currently store as a separate table `StepRunInputSchema`. `StepRunModel.input_artifacts` is the equivalent on model level.